plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java

```diff
@@ -126,4 +161,139 @@ public void testGetTableSchemaOffline()
+                transaction.commit();
+            }
+
+            // read succeeds on tables marked with Spark-specific property
+            try (Transaction transaction = newTransaction()) {
+                ConnectorSession session = newSession();
+                ConnectorMetadata metadata = transaction.getMetadata();
+                metadata.beginQuery(session);
+                ConnectorTableHandle tableHandle = getTableHandle(metadata, tableName);
+                List<ColumnHandle> columnHandles = filterNonHiddenColumnHandles(metadata.getColumnHandles(session, tableHandle).values());
+                MaterializedResult result = readTable(transaction, tableHandle, columnHandles, session, TupleDomain.all(), OptionalInt.empty(), Optional.of(ORC));
+                assertEquals(result.getRowCount(), 25); // fewer rows due to deleted file
+            }
+        }
+        finally {
+            deleteRecursively(externalLocation, RecursiveDeleteOption.ALLOW_INSECURE);
+        }
+    }
+
+    private void createExternalTable(SchemaTableName schemaTableName, HiveStorageFormat hiveStorageFormat, List<Column> columns, List<Column> partitionColumns, Optional<HiveBucketProperty> bucketProperty, Path externalLocation)
+    {
+        try (Transaction transaction = newTransaction()) {
+            ConnectorSession session = newSession();
+
+            String tableOwner = session.getUser();
+            String schemaName = schemaTableName.getSchemaName();
+            String tableName = schemaTableName.getTableName();
+
+            Table.Builder tableBuilder = Table.builder()
+                    .setDatabaseName(schemaName)
+                    .setTableName(tableName)
+                    .setOwner(tableOwner)
+                    .setTableType(TableType.EXTERNAL_TABLE.name())
+                    .setParameters(ImmutableMap.of(
+                            PRESTO_VERSION_NAME, TEST_SERVER_VERSION,
+                            PRESTO_QUERY_ID_NAME, session.getQueryId()))
+                    .setDataColumns(columns)
+                    .setPartitionColumns(partitionColumns);
+
+            tableBuilder.getStorageBuilder()
+                    .setLocation(externalLocation.toString())
+                    .setStorageFormat(StorageFormat.create(hiveStorageFormat.getSerDe(), hiveStorageFormat.getInputFormat(), hiveStorageFormat.getOutputFormat()))
+                    .setBucketProperty(bucketProperty)
+                    .setSerdeParameters(ImmutableMap.of());
+
+            PrincipalPrivileges principalPrivileges = testingPrincipalPrivilege(tableOwner, session.getUser());
+            transaction.getMetastore().createTable(session, tableBuilder.build(), principalPrivileges, Optional.of(externalLocation), true, EMPTY_TABLE_STATISTICS);
+
+            transaction.commit();
+        }
+    }
+
+    private java.nio.file.Path extractResourceZipToTemporaryDirectory(String resourceName)
```

Comment:
Looks like quite some amount of code being copied from super, why?

Reply:
I can add some helper methods to avoid this.